modules/openapi-generator/src/main/resources/Java/libraries/jersey2/model_oneof_doc.mustache

```diff
@@ -0,0 +1,38 @@
+# {{classname}}
+
+{{#description}}
+{{&description}}
+
+{{/description}}
+## oneOf schemas
+{{#oneOf}}
+* [{{{.}}}]({{{.}}}.md)
+{{/oneOf}}
```

Comment:
Let's step back. What does "combine allOf and oneOf together" really mean? Can you give an example of use case on that? What does the payload look like?

Reply:
Sure. A use case for this is when a user extracts common shared properties into a schema and they choose to have those properties stored in the oneOf schema as additionalProperties, not through allOf inheritance.
Let's look at an example:
```
EdibleInterface:
  type: object
  properties:
    calories:
      type: integer
Apple:
  type: object
  properties:
    color:
      type: string
    varietal:
      type: string
ComposedSchema:
  allOf:
    - $ref: '#/components/schemas/EdibleInterface'
  oneOf:
    - $ref: '#/components/schemas/Apple'
```
And a sample payload is:
{"calories": 110, "color": "yellow", "varietal": "golden delicious"}
At the ComposedSchema level, allOf validates + oneOf validates, and if we chose to return an instance of Apple when deserializing ComposedSchema, Apple includes the property calories as an additionalProperty.

Reply:
Let's focus on "ComposedSchema" for the time being and please forget about my implementation for the time being. Given the following:
```
ComposedSchema:
  allOf:
    - $ref: '#/components/schemas/EdibleInterface'
  oneOf:
    - $ref: '#/components/schemas/Apple'
```
oneOf and allOf are defined at the same level. Your understanding is that the properties defined in the oneOf schema are **automatically unfolded** and included in the object "ComposedSchema" (same level as the properties defined in the allOf schema), right?

Reply:
Yes, that's what I am trying to convey with the question
ComposedSchema says that a payload must validate both EdibleInterface + Apple, schemas together, correct?

Reply:
Assuming the answer to your question is yes, then the following is also valid, right?
```
ComposedSchema:
  allOf:
    - $ref: '#/components/schemas/EdibleInterface'
  oneOf:
    - type: string
    - type: integer
```
Does it make sense to you? If it does, what does the payload (e.g. JSON) look like?

Reply:
I think that it can only be combined when they are of the same type. What do you think?

Reply:
> I think that it can only be combined when they are of the same type. What do you think?

My view is that it's not up to me (or you) whether there's such a rule to ensure the same type. It comes down to the definition in JSON schema. I may have missed it but I don't see such a rule explicitly (or implicitly) defined.

One more example. What about the following:
```
ComposedSchema:
  allOf:
    - $ref: '#/components/schemas/EdibleInterface'
  oneOf:
    - $ref: '#/components/schemas/Apple'
  oneOf:
    - $ref: '#/components/schemas/Dog'
```

I assume it's also valid. Do you agree?

Reply:
Hmm isn't oneof a key? I thought that it was and that you can't have two of the same key. 
Based on that thought, I think that it is not valid. 
I will need to look at the spec to see what it says. Yup it is not up to us, I am asking about your understanding and interpretation.

Reply:
From the spec allof is defined as:
The OpenAPI Specification allows combining and extending model definitions using the allOf property of JSON Schema, in effect offering model composition. allOf takes an array of object definitions that are validated independently but together compose a single object.